### PR TITLE
test built-in suites: allow empy suite to skip

### PR DESCRIPTION
**Pull request by matthewrmshin**
... If EmPy is not available.

### CODE REVIEW

File: tests/validate/04-builtin-suites.t [hunk 1]
```diff
@@ -39,9 +39,17 @@ with open(file_name, 'r') as in_file:
 set_test_number $((( ((${#SUITES[@]})) * 2 )))
 #-------------------------------------------------------------------------------
 # Validate suites.
+NO_EMPY=true
+if python -c 'import em' 1>'/dev/null' 2>&1; then
```
Comment (hjoliver) [inline tests/validate/04-builtin-suites.t]: This test is not sufficient - see https://github.com/cylc/cylc/pull/2734#issuecomment-409001677 and preceding comments.  
@TomekTrzeciak did it this way (after modifying `cylc check-software` to do a more comprehensive test):
https://github.com/cylc/cylc/blob/master/tests/empy/00-simple.t#L20

Comment (matthewrmshin) [reply tests/validate/04-builtin-suites.t]: Fair enough. Check updated.

File: tests/validate/04-builtin-suites.t [hunk 2]
```diff
@@ -39,9 +39,17 @@ with open(file_name, 'r') as in_file:
...
 #-------------------------------------------------------------------------------
 # Validate suites.
+NO_EMPY=true
+if cylc check-software 2>'/dev/null' | grep -q '^Python:EmPy.*([^-]*)$'; then
+    NO_EMPY=false
+fi
 for suite in ${SUITES[@]}; do
     suite_name=$(sed 's/\//-/g' <<<"${suite:$ABS_PATH_LENGTH}")
     TEST_NAME="${TEST_NAME_BASE}${suite_name}"
+    if "${NO_EMPY}" && [[ "$(head -1 "${suite}")" == '#!EmPy' ]]; then
```
Comment (TomekTrzeciak) [inline tests/validate/04-builtin-suites.t]: You probably want regex matching like [this one](https://github.com/cylc/cylc/blob/master/lib/parsec/fileparse.py#L267):

```
head -1 "${suite}" | grep -q '^#![Ee]m[Pp]y'
```

Comment (matthewrmshin) [reply tests/validate/04-builtin-suites.t]: Fair enough. Updated with `grep -qi '^#!empy' < <(head -1 "${suite}")`.

**Comment by matthewrmshin** [comment]
(In case this has gone off the radar. I believe I have fixed the issues.)

**Comment by hjoliver** [comment]
Sorry, just (temporarily) over-busy at the moment.

**Comment by matthewrmshin** [comment]
(Just a test, single review approval enough.)

**Pull request merged**